### PR TITLE
Add validation for name. Fix logic in checking validation of fields

### PR DESCRIPTION
**Pull request by nepgya2803**

### CODE REVIEW

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx [hunk 1]
```diff
@@ -1,50 +1,49 @@
...
-      return false;
-    } 
-
-    else 
-    return true;
-  };
+export const validNameRegex = new RegExp("\b([A-ZÀ-ÿ][-,a-z. ']+[ ]*)+");
 
+export const validateName = (name) => {
+  return (validNameRegex.test(name) === true)  ? true : false;
```
Comment (tanngocIT) [inline bookingticketui/src/components/sign-up/sign-up-validation.jsx:10]: return validNameRegex.test(name)

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx [hunk 2]
```diff
@@ -1,50 +1,49 @@
...
-      alert("Wrong email format");
-      return false;
-    }
-
+export const validatePassword = (password) => {
+  if (
+    (password.length > 16 || password.length < 6) &&
+    validPasswordRegex.test(password) === false
+  ) {
+    alert(
```
Comment (tanngocIT) [inline bookingticketui/src/components/sign-up/sign-up-validation.jsx:18]: use toast

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx [hunk 3]
```diff
@@ -1,50 +1,49 @@
...
+      "Password length must be between 6 to 16 words. It must contain at least 1 uppercase, 1 special symbol, and 1 number"
+    );
+
+    return false;
+  } else return true;
+};
+
+export const validateEmail = (email) => {
+  if (email == null) {
+    alert("Email is not allow to be empty");
```
Comment (tanngocIT) [inline bookingticketui/src/components/sign-up/sign-up-validation.jsx:28]: use toast

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx [hunk 4]
```diff
@@ -1,50 +1,49 @@
...
+    return false;
+  }
+
+  if (validator.isEmail(email) === false) {
+    alert("Wrong email format");
```
Comment (tanngocIT) [inline bookingticketui/src/components/sign-up/sign-up-validation.jsx:33]: use toast

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx [hunk 5]
```diff
@@ -1,50 +1,49 @@
...
-
-    else
-    {
-      alert("Confirm password must be the same as password");
-      return false;
-    }
-  };
-
+  } else {
+    alert("Confirm password must be the same as password");
```
Comment (tanngocIT) [inline bookingticketui/src/components/sign-up/sign-up-validation.jsx:46]: use toast

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx (hunk 1 above)
Comment (ngqtrieu99) [reply bookingticketui/src/components/sign-up/sign-up-validation.jsx]: fixed

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx (hunk 2 above)
Comment (ngqtrieu99) [reply bookingticketui/src/components/sign-up/sign-up-validation.jsx]: fixed

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx (hunk 3 above)
Comment (ngqtrieu99) [reply bookingticketui/src/components/sign-up/sign-up-validation.jsx]: fixed

Comment (ngqtrieu99) [reply bookingticketui/src/components/sign-up/sign-up-validation.jsx]: fixed

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx (hunk 4 above)
Comment (ngqtrieu99) [reply bookingticketui/src/components/sign-up/sign-up-validation.jsx]: fixed

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx (hunk 5 above)
Comment (ngqtrieu99) [reply bookingticketui/src/components/sign-up/sign-up-validation.jsx]: fixed

File: bookingticketui/src/components/sign-up/sign-up-validation.jsx [hunk 6]
```diff
@@ -1,50 +1,49 @@
...
-    
-    if (validator.isEmail(email) === false) {
-      alert("Wrong email format");
-      return false;
-    }
-
+export const validatePassword = (password) => {
+  if (
+    (password.length > 16 || password.length < 6) &&
+    validPasswordRegex.test(password) === false
```
[the remark this answers is not in the document]
Comment (ngqtrieu99) [reply bookingticketui/src/components/sign-up/sign-up-validation.jsx]: fixed

**Pull request merged**